lib/classes/ConfigSchemaHandler/normalizeAjvErrors.test.js

```diff
@@ -0,0 +1,563 @@
+                          },
+                        },
+                        required: ['http'],
+                        additionalProperties: false,
+                      },
+                    ],
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    };
+
+    const validate = ajv.compile(schema);
+    validate(userConfig);
+    ajvErrors = validate.errors;
+    expect(normalizeAjvErrors(ajvErrors, userConfig).map(err => err.message)).to.deep.equal(
+      expectedErrorMessages
+    );
+  });
+
+  it('should show error messages for unsupported function name', () => {
+    userConfig = {
+      functions: {
+        $omeFunc: {
+          handler: 'handler.main',
+        },
+      },
+    };
+    schema = {
+      type: 'object',
+      properties: {
+        functions: {
+          type: 'object',
+          patternProperties: {
+            '^[a-zA-Z0-9-_]+$': {
+              type: 'object',
+            },
+          },
+          additionalProperties: false,
+        },
+      },
+    };
+    const validate = ajv.compile(schema);
+    validate(userConfig);
+    ajvErrors = validate.errors;
+    expect(normalizeAjvErrors(ajvErrors, userConfig).map(err => err.message)).to.deep.equal([
+      "Function name '$omeFunc' must be alphanumeric at functions.",
+    ]);
+  });
+
+  it('should show both errors from separate different functions ', () => {
+    userConfig = {
+      functions: {
+        broken$Func: {
+          events: [{ http: 'get /foo' }],
+        },
+        someFunc: {
+          events: [{ httpp: 'get /foo' }],
+        },
+      },
+    };
+
+    const expectedErrorMessages = [
+      "Function name 'broken$Func' must be alphanumeric at functions.",
+      "Unsupported function event 'httpp' at functions.someFunc.events[0]",
+    ];
+
+    schema = {
+      type: 'object',
+      properties: {
+        functions: {
+          type: 'object',
+          patternProperties: {
+            '^[a-zA-Z0-9-_]+$': {
+              type: 'object',
+              properties: {
+                events: {
+                  type: 'array',
+                  items: {
+                    anyOf: [
+                      {
+                        type: 'object',
+                        properties: {
+                          http: { type: 'string' },
+                        },
+                        required: ['http'],
+                        additionalProperties: false,
+                      },
+                    ],
+                  },
+                },
+              },
+            },
+          },
+          additionalProperties: false,
+        },
+      },
+    };
+
+    const validate = ajv.compile(schema);
+    validate(userConfig);
+    ajvErrors = validate.errors;
+    expect(normalizeAjvErrors(ajvErrors, userConfig).map(err => err.message)).to.deep.equal(
+      expectedErrorMessages
+    );
+  });
+});
+
+describe('#groupAjvErrors', () => {
```

Comment:
This tests characteristics of internal implementation, I wouldn't do that (we may decide on some other internal implemetation, without changing input/output contract), and that should not require rewrite of tests.

So I would ditch this group of tests (and if it covers well some cases not well covered above, I would improve above section so they're covered with natural input/output tests)

Reply:
I'll remove those tests then. I added them more like an example of how `groupAjvErrors` function behaves (as I remember you asked for such example). 

I'll remove these group of tests then. I'll make it cleaner